### DISCUSSION TITLE
Rootless runtime support

### DISCUSSION THREAD

**Discussion by gnodar01** [bilayer-containers/bilayers#63] [Ideas]
I attended the Dl4MicEverywhere workshop, which was great. They cover a different use case than us, but share a lot of the same ideas in terms of approach. One of the things they do similar to us is taking they put each model in docker images, and have an automated CI/CD system to do the building.

Like us, they also then ask rely on their users using docker desktop to run those images. During the workshop, several people raised concerns about the reliance on docker specifically.

It had not occurred to me that this would be a problem. I knew there were concerns around Docker requiring sudo access, but since that is only actually necessary at build time, which we do ourselves, and not running, I assumed it would not be a problem.

Apparently, it is a problem. Several institutes will ban Docker outright.

In theory, we are building OCI compliant containers, and so we should already have complete Singularity support. We should test this though. If true, we won't have to modify any existing code, we just have to add information in the docs on how to set up Singularity to run Bilayers images.

**Comment by bethac07** [comment]
I spent a bit of time looking into this earlier this week after one of the other Docker-y workshops - there was at one point a Singularity Desktop for Mac, but it wasn't maintained and has been yanked. You CAN install Singularity for Windows and Mac, but it's [not trivial](https://docs.sylabs.io/guides/3.2/user-guide/installation.html#install-on-windows-or-mac) and breaks our "you don't need the command line" promise. 

I haven't yet started playing, but there is a [Podman Desktop](https://podman-desktop.io/), that may end up being a better fit for our needs, but more investigation needed.

**Comment by gnodar01** [comment]
Ah, okay. Changed title to be more in line with what IT people actually want.

**Comment by bethac07** [comment]
Closed as we fully support Podman.
